lib/ansible/plugins/callback/telegraf.py

```diff
@@ -0,0 +1,290 @@
+      - pip install pytelegraf (udp) or pytelegraf[http]
+      - running telegraf reachable by ansible controller configured with desired output to a metrics collector
+      - Tower http://docs.ansible.com/ansible-tower/latest/html/administration/tipsandtricks.html#using-callback-plugins-with-tower
+      - if running ansible < 2.4.0 you may need to remove the ini sections from this doc blob and use ENV vars
+    short_description: Sends play events to a telegraf forwarder (to metrics collector)
+    version_added: "2.4"
+    description:
+        - This is an ansible callback plugin that sends playbook stats to a telegraf listener input (udp/http) during playbook execution.
+        - Configure with <playbook root>callback_plugins/ansible-telegraf-callback/telegraf_callback.yml
+        - or using the module args or options in ansible.cfg
+    troubleshooting:
+      - it's difficult to debug udp connections, so start with http
+      - run with -vvvv.  In the first few lines you should see the callback is loaded
+      - grep --binary-file=text changed /var/log/telegraf.log to see if telegraf is receiving 
+      - ensure telegraf is running with correct input blocks in /etc/telegraf/telegraf.d/
+        inputs-socket-listener.conf or inputs-http-listener.conf https://github.com/influxdata/telegraf/tree/master/plugins/inputs/socket_listener
+    options:
+      host:
+        required: True
+        description: telegraf host reachable from the ansible controller
+        env:
+          - name: ANSIBLE_TELEGRAF_CALLBACK_HOST
+      metric_prefix:
+        description: metrics prefix for all metrics sent to telegraf.
+        env:
+          - name: ANSIBLE_TELEGRAF_CALLBACK_METRIC_PREFIX
+        default: "ansible."
+        ini:
+          - section: callback_telegraf
+            key: metric_prefix
+      metric_per_host:
+        description: summarize over all tasks per host
+        env:
+          - name: ANSIBLE_TELEGRAF_CALLBACK_METRIC_PER_HOST
+        default: True
+        ini:
+          - section: callback_telegraf
+            key: metric_per_host
+      metric_summary:
+        description: summarize stats over all hosts.
+        env:
+          - name: ANSIBLE_TELEGRAF_CALLBACK_METRIC_SUMMARY
+        default: False
+        ini:
+          - section: callback_telegraf
+            key: metric_summary
+      wire_protocol:
+        description: http, https or udp.  If http you must pip install pytelegraf[http]. Default udp
+        choices: ['http', 'https', 'udp']
+        env:
+          - name: ANSIBLE_TELEGRAF_CALLBACK_WIRE_PROTOCOL
+        default: http
+        ini:
+          - section: callback_telegraf
+            key: wire_protocol
+'''
+
+import getpass
+import os.path
+import socket
+from datetime import datetime, timedelta
+import yaml
+
+
+try:
+    # Ansible v2
+    from ansible.plugins.callback import CallbackBase
+    from __main__ import cli
```

Comment:
~you don't seem to use 'cli' why import it?
also the base class does this already.~ v1 compatibility it seems.

Reply:
removing cli.  v2 support only is fine.